### PR TITLE
Add k8s-staging-cluster-addons staging job

### PR DESCRIPTION
**Pull request by justinsb**
This builds images for the cluster-addons repo.

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-cluster-addons.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-cluster-lifecycle-cluster-addons
+      decorate: true
+      branches:
+        - ^master$
+        - ^release-.*
+        # Build tags also
+        # this is a regex for semver, from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+        - ^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
```
Comment (dims) [inline config/jobs/image-pushing/k8s-staging-cluster-addons.yaml:13]: that's a wicked regex!

image

Comment (justinsb) [reply config/jobs/image-pushing/k8s-staging-cluster-addons.yaml:13]: It sure is ... I copied and pasted it from someone else, who copied and pasted it from someone else, etc until we get to someone who was probably just trolling :-)

Comment (dims) [reply config/jobs/image-pushing/k8s-staging-cluster-addons.yaml:13]: LOL +1

**Comment by justinsb** [comment]
The PR that added the cloudbuild job merged ... :crossed_fingers: 

/hold cancel

**Pull request merged**